packages/anthropic-vertex/src/anthropic-vertex-provider.ts

```diff
@@ -0,0 +1,174 @@
+     */
+  headers?: Record<string, string>;
+
+  /**
+Custom fetch implementation. You can use it as a middleware to intercept requests,
+or to provide a custom fetch implementation for e.g. testing.
+    */
+  fetch?: typeof fetch;
+
+  generateId?: () => string;
+}
+
+/**
+Create an Anthropic provider instance.
+ */
+export function createAnthropicVertex(
+  options: AnthropicVertexProviderSettings = {},
+): AnthropicVertexProvider {
+
+  const config = {
+    projectId: loadSetting({
+      settingValue: options.projectId,
+      settingName: 'projectId',
+      environmentVariableName: 'GOOGLE_VERTEX_PROJECT_ID',
+      description: 'Google Vertex project id',
+    }),
+    region: loadSetting({
+      settingValue: options.region,
+      settingName: 'region',
+      environmentVariableName: 'GOOGLE_VERTEX_REGION',
+      description: 'Google Vertex region',
+    }),
+    googleAuth: options.googleAuth,
+  };
+
+  if (!config.region) {
+    throw new Error(
+      'No region was given. The client should be instantiated with the `region` option or the `GOOGLE_VERTEX_REGION` environment variable should be set.',
+    );
+  }
+  
+  if(!config.projectId) {
+    throw new Error(
+      'No project was given. The client should be instantiated with the `projectID` option or the `GOOGLE_VERTEX_PROJECT_ID` environment variable should be set.',
+    );
+  }
+
+  const baseURL =
+    withoutTrailingSlash(options.baseURL) ??
+    `https://${config.region}-aiplatform.googleapis.com/v1`;
+
+  const auth =
+    options.googleAuth ?? new GoogleAuth({ scopes: 'https://www.googleapis.com/auth/cloud-platform' });
+
+  // const token = await auth.getAccessToken()
+
+  const DEFAULT_VERSION = 'vertex-2023-10-16';
+
+  // const getHeaders = () => ({
+  //   'x-api-key': loadApiKey({
+  //     apiKey: options.apiKey,
+  //     environmentVariableName: 'ANTHROPIC_API_KEY',
+  //     description: 'Anthropic',
+  //   }),
+  //   ...options.headers,
+  // });
+// 
+
+  const createChatModel = (
+    modelId: AnthropicMessagesModelId,
+    settings: AnthropicMessagesSettings = {},
+  ) =>
+    new AnthropicMessagesLanguageModel(modelId, settings, {
+      provider: 'anthropic.messages',
+      baseURL,
+      headers: () => ({
+        ...options.headers
+      }),
+      fetch: options.fetch,
+      projectId: config.projectId,
+      region: config.region,
+      googleAuth: auth
+    });
+
+  const provider = function (
+    modelId: AnthropicMessagesModelId,
+    settings?: AnthropicMessagesSettings,
+  ) {
+    if (new.target) {
+      throw new Error(
+        'The Anthropic model function cannot be called with the new keyword.',
+      );
+    }
+
+    return createChatModel(modelId, settings);
+  };
+
+  provider.languageModel = createChatModel;
+  provider.chat = createChatModel;
+
+  return provider as AnthropicVertexProvider;
+}
+
+/**
+Default Anthropic provider instance.
+ */
+export const anthropicVertex = createAnthropicVertex();
```

Comment:
This line is creates a bug:

When this module gets evaluated `createAnthropicVertex();` gets called.

The default `anthropicVertex` provider then cannot get created with an empty `options` property, because `loadSetting` gets called with environment variables set and will complain that those variables are do not get return a valid value, after falling through, see here -> https://github.com/vercel/ai/blob/main/packages/provider-utils/src/load-setting.ts#L42

This produces an error, even if `createAnthropicVertex()` gets called with a correct `options` property later, practically excluding this method to set `region`, `projectID` and other options programmatically.

It took me a while to trace this error message back to the line mentioned above ->
```
/Users/fry/code-local/ai/node_modules/.pnpm/@ai-sdk+provider-utils@1.0.16_zod@3.23.8/node_modules/@ai-sdk/provider-utils/src/load-setting.ts:45
    throw new LoadSettingError({
          ^


LoadSettingError [AI_LoadSettingError]: Google Vertex project id setting is missing. Pass it using the 'projectId' parameter or the GOOGLE_VERTEX_PROJECT_ID environment variable.
    at loadSetting (/Users/fry/code-local/ai/node_modules/.pnpm/@ai-sdk+provider-utils@1.0.16_zod@3.23.8/node_modules/@ai-sdk/provider-utils/src/load-setting.ts:45:11)
    at createAnthropicVertex (/Users/fry/code-local/ai/packages/anthropic-vertex/src/anthropic-vertex-provider.ts:91:16)
    at <anonymous> (/Users/fry/code-local/ai/packages/anthropic-vertex/src/anthropic-vertex-provider.ts:177:32)
    at ModuleJob.run (node:internal/modules/esm/module_job:262:25)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:482:26)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:117:5) {
  cause: undefined,
  [Symbol(vercel.ai.error)]: true,
  [Symbol(vercel.ai.error.AI_LoadSettingError)]: true
}
```

Reply:
How did this got resolved?